lib/process-manager.ts

```diff
@@ -57,40 +60,88 @@ class RawSubprocessStream extends Streams.ObjectReadWriteStream<string> {
 		this.process = process;
 	}
 	_write(message: string) {
-		if (!this.process.getProcess().connected) {
+		if (!this.process.isConnected) {
 			// no error because the crash handler should already have shown an error, and
 			// sometimes harmless messages are sent during cleanup
 			return;
 		}
-		this.process.process.send(message);
+		this.process._send(message);
 		// responses are handled in ProcessWrapper
 	}
 }
 
-interface ProcessWrapper {
-	load: number;
-	process: ChildProcess | Worker;
-	release: () => Promise<void>;
-	getProcess: () => ChildProcess;
+abstract class ProcessWrapper {
+	process: ChildProcess | null = null;
+	clusterWorker: cluster.Worker | null = null;
+	worker: worker_threads.Worker | null = null;
```

Comment:
These types are imprecise (all could be `null`) forcing you to bypass the typechecker later with the `!` operator. Can avoid this by using a union of the form `{type: 'process', data: ChildProcess} | {type: 'worker', data: worker_threads.Worker} | ...`

Reply:
I did consider that but in general I don't like having worse performance to satisfy the typechecker.

Reply:
You could define `ProcessWrapper` externally to have that type, but most of the assertions are inside `ProcessWrapper` so it wouldn't help much anyway.